### PR TITLE
docs: Add preconditions to some samples

### PR DESCRIPTION
**Pull request by JesseLovelace**
Adds preconditions to some Java samples. This is half the samples that need to be added, @sydney-munro is going to do the other half

### CODE REVIEW

File: samples/snippets/src/main/java/com/example/storage/object/ComposeObject.java [hunk 1]
```diff
@@ -46,12 +46,22 @@ public static void composeObject(
...
     Storage storage = StorageOptions.newBuilder().setProjectId(projectId).build().getService();
 
+    // Optional: set a generation-match precondition to avoid potential race
+    // conditions and data corruptions. The request returns a 412 error if the
+    // preconditions are not met.
+    // For a target object that does not yet exist, set the DoesNotExist precondition.
+    Storage.BlobTargetOption precondition = Storage.BlobTargetOption.doesNotExist();
+    // If the destination already exists in your bucket, instead set a generation-match
+    // precondition:
+    // Storage.BlobTargetOption precondition = Storage.BlobTargetOption.generationMatch();
```
Comment (Shabirmean) [inline samples/snippets/src/main/java/com/example/storage/object/ComposeObject.java:56]:
```suggestion
    // preconditions are not met.
    
    // For a target object that does not yet exist, set the DoesNotExist precondition.
    Storage.BlobTargetOption precondition = Storage.BlobTargetOption.doesNotExist();
    
    // If the destination already exists, instead set a generation-match precondition:
    // Storage.BlobTargetOption precondition = Storage.BlobTargetOption.generationMatch();
```

- Add a empty line after line 51 on line 52 and also on line 55
- Make the comment on line 54 to be on single line

File: samples/snippets/src/main/java/com/example/storage/object/CopyObject.java [hunk 2]
```diff
@@ -34,14 +32,25 @@ public static void copyObject(
...
     // The ID of the bucket to copy the object to
-    // String targetBucketName = "target-object-bucket"
+    // String targetBucketName = "target-object-bucket";
 
     Storage storage = StorageOptions.newBuilder().setProjectId(projectId).build().getService();
-    Blob blob = storage.get(sourceBucketName, objectName);
+    BlobId source = BlobId.of(sourceBucketName, objectName);
+    BlobId target =
+        BlobId.of(
+            targetBucketName, objectName); // you could change "objectName" to rename the object
```
Comment (Shabirmean) [inline samples/snippets/src/main/java/com/example/storage/object/CopyObject.java:41]: Any particular reason why you split it into multiple lines? Can we have it all in one line?

```java
BlobId target = BlobId.of(targetBucketName, objectName);
```

File: samples/snippets/src/main/java/com/example/storage/object/CopyObject.java [hunk 3]
```diff
@@ -34,14 +32,25 @@ public static void copyObject(
...
 
-    // This keeps the original name, you could also do
-    // copyTo(targetBucketName, "target-object-name") to change the name
-    blob.copyTo(targetBucketName);
+    // Optional: set a generation-match precondition to avoid potential race
+    // conditions and data corruptions. The request returns a 412 error if the
+    // preconditions are not met.
+    // For a target object that does not yet exist, set the DoesNotExist precondition.
```
Comment (Shabirmean) [inline samples/snippets/src/main/java/com/example/storage/object/CopyObject.java:46]: Add empty lines as suggested above.

File: samples/snippets/src/main/java/com/example/storage/object/CopyOldVersionOfObject.java [hunk 4]
```diff
@@ -44,10 +44,20 @@ public static void copyOldVersionOfObject(
     // String newObjectName = "your-new-object";
 
     Storage storage = StorageOptions.newBuilder().setProjectId(projectId).build().getService();
+
+    // Optional: set a generation-match precondition to avoid potential race
+    // conditions and data corruptions. The request returns a 412 error if the
+    // preconditions are not met.
+    // For a target object that does not yet exist, set the DoesNotExist precondition.
```
Comment (Shabirmean) [inline samples/snippets/src/main/java/com/example/storage/object/CopyOldVersionOfObject.java:51]: (Suggestion) to add empty lines as showed in earlier comment

File: samples/snippets/src/main/java/com/example/storage/object/MoveObject.java [hunk 5]
```diff
@@ -45,13 +42,25 @@ public static void moveObject(
...
+    BlobId source = BlobId.of(sourceBucketName, sourceObjectName);
+    BlobId target = BlobId.of(targetBucketName, targetObjectName);
+
+    // Optional: set a generation-match precondition to avoid potential race
+    // conditions and data corruptions. The request returns a 412 error if the
+    // preconditions are not met.
+    // For a target object that does not yet exist, set the DoesNotExist precondition.
+    Storage.BlobTargetOption precondition = Storage.BlobTargetOption.doesNotExist();
+    // If the destination already exists in your bucket, instead set a generation-match
+    // precondition:
```
Comment (Shabirmean) [inline samples/snippets/src/main/java/com/example/storage/object/MoveObject.java:54]: (Suggestion) to add line break as shown in earlier comment

File: samples/snippets/src/main/java/com/example/storage/object/UploadEncryptedObject.java [hunk 6]
```diff
@@ -48,10 +48,21 @@ public static void uploadEncryptedObject(
...
+    // If the destination already exists in your bucket, instead set a generation-match
+    // precondition:
+    // Storage.BlobTargetOption precondition = Storage.BlobTargetOption.generationMatch();
+
     storage.create(
         blobInfo,
         Files.readAllBytes(Paths.get(filePath)),
-        Storage.BlobTargetOption.encryptionKey(encryptionKey));
+        Storage.BlobTargetOption.encryptionKey(encryptionKey),
+        precondition);
```
Comment (Shabirmean) [inline samples/snippets/src/main/java/com/example/storage/object/UploadEncryptedObject.java:65]: (nit) Move this to the same line (L64)

File: samples/snippets/src/main/java/com/example/storage/object/UploadKmsEncryptedObject.java [hunk 7]
```diff
@@ -45,7 +45,17 @@ public static void uploadKmsEncryptedObject(
...
-    storage.create(blobInfo, data, Storage.BlobTargetOption.kmsKeyName(kmsKeyName));
+
+    // Optional: set a generation-match precondition to avoid potential race
+    // conditions and data corruptions. The request returns a 412 error if the
+    // preconditions are not met.
+    // For a target object that does not yet exist, set the DoesNotExist precondition.
+    Storage.BlobTargetOption precondition = Storage.BlobTargetOption.doesNotExist();
+    // If the destination already exists in your bucket, instead set a generation-match
+    // precondition:
+    // Storage.BlobTargetOption precondition = Storage.BlobTargetOption.generationMatch();
```
Comment (Shabirmean) [inline samples/snippets/src/main/java/com/example/storage/object/UploadKmsEncryptedObject.java:56]: Add line breaks to make the comments clearer

**Review by Shabirmean — COMMENTED** [review]
LGTM Overall.

Some minor formatting related comments. Also PTAL at the failing tests

File: samples/snippets/src/main/java/com/example/storage/object/ComposeObject.java (hunk 1 above)
Comment (JesseLovelace) [reply samples/snippets/src/main/java/com/example/storage/object/ComposeObject.java:56]: > Add a empty line after line 51 on line 52 and also on line 55

I'm not sure I agree with that, it's all conceptually linked, as "the part of the sample that's about preconditions" so I think it makes more sense as one block

>Make the comment on line 54 to be on single line

That comment was originally written as a single line, unfortunately Owlbot [auto-edits](https://github.com/googleapis/java-storage/pull/1600/commits/4eded73d48a6c47afea6a6f16776f4d5cb048273#diff-62e96d6add7b3154d4d872ae825a6901398b3f5c8e9ed77d1cf8b29d0b669960R54) all PRs to conform to certain style guidelines including a maximum amount of characters per line and there's no real way to fight with it

File: samples/snippets/src/main/java/com/example/storage/object/CopyObject.java (hunk 2 above)
Comment (JesseLovelace) [reply samples/snippets/src/main/java/com/example/storage/object/CopyObject.java:41]: It was originally written on one line, [Owlbot changed it](https://github.com/googleapis/java-storage/pull/1600/commits/4eded73d48a6c47afea6a6f16776f4d5cb048273#diff-de4371495edfc135eac9d6142b6a3c480b3e2593604c0d0452dd1fce28fa67d7R39) and will just change it back if i try to fix it

File: samples/snippets/src/main/java/com/example/storage/object/UploadEncryptedObject.java (hunk 6 above)
Comment (JesseLovelace) [reply samples/snippets/src/main/java/com/example/storage/object/UploadEncryptedObject.java:65]: [Owlbot did this](https://github.com/googleapis/java-storage/pull/1600/commits/4eded73d48a6c47afea6a6f16776f4d5cb048273#diff-9ad60db022b5fd58482b3e639195b3b8dd7d6d320020ff82739d2506b96ac834R57) and will just change it back if I try to fix it

**Review by BenWhitehead — APPROVED** [review]
Thanks for adding the null checks!

**Review by Shabirmean — APPROVED** [review]

**Pull request merged**